src/Strategy.sol

```diff
@@ -0,0 +1,616 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity ^0.8.24;
+
+import {
+    AccessControlUpgradeable,
+    Address,
+    ERC20PermitUpgradeable,
+    ERC20Upgradeable,
+    IERC20,
+    IERC20Metadata,
+    Math,
+    ReentrancyGuardUpgradeable,
+    SafeERC20
+} from "./Common.sol";
+
+import {IVault} from "src/interface/IVault.sol";
+import {IStrategy} from "src/interface/IStrategy.sol";
+import {IProvider} from "src/interface/IProvider.sol";
+import {Guard} from "src/module/Guard.sol";
+
+contract Strategy is IVault, ERC20PermitUpgradeable, AccessControlUpgradeable, ReentrancyGuardUpgradeable {
```

Comment:
since most of the logic is the same as Vault, should we have a BaseVault 

that they both inherit?

Reply:
created BaseVault